### DISCUSSION TITLE
Backgrounds are cut off at the top/bottom and sides at non 16:9 aspect ratios

### DISCUSSION THREAD

**Discussion by tovc** [ppy/osu#22534] [Ideas] [answered]
this is the full background at 1920x1080
![osu_2023-02-06_17-00-26](https://user-images.githubusercontent.com/49025619/216929151-33e6eb9a-e876-4ebe-8999-850ad9423807.jpg)

this is an extreme example, but it shows the edges being cut off. this issue is present (but less prominent) on 4:3 aspect ratios aswel
![osu_2023-02-06_16-59-43](https://user-images.githubusercontent.com/49025619/216929303-b12d44a5-bdd3-4b24-a4f9-a6556956b716.jpg)

this is a more realistic example, and one that affects me, as an ultrawide user
![osu_2023-02-06_16-59-54](https://user-images.githubusercontent.com/49025619/216929556-e7d95872-5cd2-4085-900d-d669bff5d92d.jpg)

in osu!stable, this issue can be (somewhat) remedied by simply running the game at a resolution with a 16:9 aspect ratio, then ticking the 'render at native resolution'. this solution is far from ideal, but it's what I personally opt for
![Screenshot (5)](https://user-images.githubusercontent.com/49025619/216930346-30951f11-b57e-4cc5-ac50-ca7198a5fab5.png)

no such solution exists for lazer. I'd personally suggest letterboxing/pillarboxing the backgrounds, then blurring the blank space

**Comment by bdach** [comment]
So you're saying you *want* letterboxing on backgrounds? I'd say the opposite personally.

**Reply by tovc** [reply]
Personal preference, but yeah. I'm certainly not saying that it'd be for everyone, so it'd need to be an option.

Currently, the only way to avoid backgrounds being cut off, is by using windowed mode, which adds enough input latency to be uncomfortable for gameplay.

**Comment by peppy** [comment, marked as answer]
You can do this using a screen scaling mode of your choice:

https://user-images.githubusercontent.com/191335/217178899-ce1741d1-e91c-4ed5-b7d0-e207c2ac61b6.mp4

**Reply by tovc** [reply]
I wasn't aware of that, thanks!

This does work for the maps that require it, but it is moderately obtuse for actual gameplay; would it be possible/realistic to add a scaling option for backgrounds only? ie: a scaling mode that excludes hud elements, but scales the background identically to the "Excluding overlays" option

**Reply by peppy** [reply]
That's a bit weird, I don't think it should affect gameplay that much if at all?

**Reply by tovc** [reply]
perhaps I'm describing it badly; this is how it looks with the "Excluding overlays" mode
![image](https://user-images.githubusercontent.com/49025619/217238265-ef66ba07-c3c3-47a9-b9be-87e1f396ff90.png)

this is how I'd prefer it to look. this is just a messy hack done by changing the `FillMode` from `FillMode.Fill` to `FillMode.Fit` in `Background.cs`. the black pillars are just a side effect and aren't necessarily intentional. I'd personally suggest they simply be the background, set to `FillMode.Fill`, and blurred
![image](https://user-images.githubusercontent.com/49025619/217238690-57e13214-94a8-480f-95ca-1182824d78d3.png)

**Reply by peppy** [reply]
Fair enough. I think we could probably expose a fill mode for beatmap backgrounds which does what you want (with options of Fit and Fill, or just a checkbox being like "Fill screen with beatmap background").

I'd also agree to adding a colour/blur layer in the background to fill in the letterboxing. If we can do that efficiently.
